atomix/cluster/src/main/java/io/atomix/raft/zeebe/ZeebeLogAppender.java

```diff
@@ -71,5 +71,16 @@ void appendEntry(
      * @param error the error that occurred
      */
     void onCommitError(Indexed<ZeebeEntry> indexed, Throwable error);
+
+    /**
+     * Called before writing to validated that the entry to be appended will not make the log
+     * inconsistent.
+     *
+     * @param lastPosition the highest position of the previous entry
+     * @param entry current to append
+     * @return a ValidationResult containing the validation result and an error message, if it
+     *     failed
+     */
+    ValidationResult validatePositions(long lastPosition, ZeebeEntry entry);
```

Comment:
Still I think this doesn't need to be part of the Listener.

I think what we can do is to create a Validator on Broker start and put it on the Atomix configuration, then it is added to the RaftContext and can be used in the LeaderAppender. There is no need that this logic is here, since it just checks the incoming parameters no other context is needed.

Should I prepare something to show that as example?!

Reply:
This would make the other code more cleaner since you dont need to implement the method everywhere else

Reply:
> Should I prepare something to show that as example?!

No, I think I got it.